src/NLog/Targets/DatabaseTarget.cs

```diff
@@ -736,7 +727,7 @@ protected string BuildConnectionString(LogEventInfo logEvent)
                 sb.Append("User id=");
                 sb.Append(dbUserName);
                 sb.Append(";Password=");
-                var password = GetRenderedAndEscapedPassword(logEvent);
+                var password = _dbPassword.Render(logEvent);
```

Comment:
@snakefoot 
AFAIK I can't prevent calling `DbPassword.Render(logEvent);` here, expect by unit tests. Or do you know an method for that?

Reply:
Not sure I understand. Are the unit-tests performing white-box testing?

Reply:
Nope, but still I prefer compile time checks over unit tests ;)